src/com/google/enterprise/adaptor/Dashboard.java

```diff
@@ -252,6 +253,19 @@ public Object run(List request) {
   }
 
   static class JavaVersionStatusSource implements StatusSource {
+    /** Support levels for different Java versions. */
+    private enum Supported { NO, UNKNOWN, PARTIAL, YES };
+
+    /** An ordered map of Java versions to support levels. */
+    private static final Map<String, Supported> versions
+        = new LinkedHashMap<String, Supported>();
+
+    static {
+      versions.put("1.7.0_9", Supported.PARTIAL);
```

Comment:
The doc says:

"""
 If you want to use the DH (Diffie-Hellman) style of encryption and you are running the GSA
with 2048-bit encryption, minimum JRE 1.7u80 or 1.8.0_20 is required.
"""

How large of a population uses DH and runs GSA with 2048 encryption? Can we detect that and give warning then and only then?